Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt

```diff
@@ -7,71 +7,71 @@ import android.widget.*
 import com.aliucord.Constants.PLUGIN_DEVELOPMENT_CHANNEL_ID
 import com.aliucord.Constants.PLUGIN_REQUESTS_CHANNEL_ID
 import com.aliucord.Utils
+import com.aliucord.api.SettingsAPI
 import com.aliucord.entities.Plugin
 import com.aliucord.fragments.ConfirmDialog
 import com.aliucord.patcher.Hook
 import com.aliucord.patcher.Patcher
-import com.discord.databinding.WidgetChatInputBinding
+import com.aliucord.settings.delegate
 import com.discord.widgets.chat.input.ChatInputViewModel
 import com.discord.widgets.chat.input.WidgetChatInput
 import com.lytefast.flexinput.R
 
 @SuppressLint("SetTextI18n")
 internal class SupportWarn : Plugin(Manifest("SupportWarn")) {
-    private val bindingMethod = WidgetChatInput::class.java.getDeclaredMethod("getBinding").apply { isAccessible = true }
-
-    private val channelList = listOf(
-        811255667469582420L, // #offtopic
-        PLUGIN_DEVELOPMENT_CHANNEL_ID, // #plugin-development
-        868419532992172073L, // #theme-development
-        865188789542060063L, // #related-development
-        811262084968742932L, // #core-development
-        PLUGIN_REQUESTS_CHANNEL_ID,
-    )
-
-    private val chatWrapId = Utils.getResId("chat_input_wrap", "id")
-    private val gateButtonTextId = Utils.getResId("chat_input_member_verification_guard_text", "id")
-    private val gateButtonImageId = Utils.getResId("chat_input_member_verification_guard_icon", "id")
-    private val gateButtonArrowId = Utils.getResId("chat_input_member_verification_guard_action", "id")
-    private val gateButtonLayoutId = Utils.getResId("guard_member_verification", "id")
+    private val SettingsAPI.acceptedPrdNotRequests: Boolean by settings.delegate(false)
```

Comment:
why on settings instead of simply on class?

Reply:
Prettier